frontend/src/ui/section/event-detail/EventDetail.tsx

```diff
@@ -1,62 +1,109 @@
+import { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
 
-import { eventMock } from '@/mocks/event';
 import EventInfo from '@/ui/components/atoms/event-card/EventCard';
-import EventDescription from '@/ui/components/atoms/event-description/EventDescription';
 import { IconWithText } from '@/ui/components/atoms/icon-with-text/IconWithText';
+import Section from '@/ui/components/molecules/section/Section';
 import calendarClockImage from '@assets/icons/Calendar_clock_24.svg';
 import locationImage from '@assets/icons/Location_24.svg';
-import petsImage from '@assets/icons/Pets_24.svg';
+import DOMPurify from 'dompurify';
+import { EmojiConvertor } from 'emoji-js';
+
+import { getEventDetails } from '@/modules/events/application/get-details/getEventDetails';
+import { DetailedEvent } from '@/modules/events/domain/Event';
+import { createEventRepository } from '@/modules/events/infrastructure/EventRepository';
 
 import '@/ui/section/event-detail/EventDetail.scss';
 
 export function EventDetail() {
   const { id } = useParams();
-  const {
-    title,
-    month,
-    day,
-    weekday,
-    location,
-    description,
-    time,
-    petsAllowed,
-  } = eventMock;
-
-  // TODO: When backend is ready check if event exist or not and modify this condition
+  const [eventDetails, setEventDetails] = useState<DetailedEvent>();
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const eventRepository = createEventRepository();
+
+  const emojiConvertor = new EmojiConvertor.EmojiConvertor();
```

Comment:
Now its not necessary, but maybe could be good idea create a helper or hook for that in the future if we think to reuse it in more places

Reply:
Correct, I will work on it with the test. Leaving a comment in the code